### PR TITLE
Fix/did peer generation

### PR DESCRIPTION
**Pull request by forkimenjeckayang**
This PR adjusts the required chnages to have the did:peer:generator work for our use case, it intoduces
- New algorithm for key genration (X25519) keys
- Tests refinement
Closes ADORSYS-GIS/wallet-vc-libs#50

### CODE REVIEW

File: packages/multiple-did-identities/src/__tests__/generate-key-pair.spec.ts [hunk 1]
```diff
@@ -46,3 +50,115 @@ describe('generateKeyPairs', () => {
...
+
+    // Raw keys should be unique
+    expect(firstKeyPair.rawPublicKey).not.toEqual(secondKeyPair.rawPublicKey);
+    expect(firstKeyPair.rawPrivateKey).not.toEqual(secondKeyPair.rawPrivateKey);
+
+    // JWK representations should also be unique
+    expect(firstKeyPair.publicKeyJwk).not.toEqual(secondKeyPair.publicKeyJwk);
+    expect(firstKeyPair.privateKeyJwk).not.toEqual(secondKeyPair.privateKeyJwk);
+  });
+});
```
Comment (Ogenbertrand) [inline packages/multiple-did-identities/src/__tests__/generate-key-pair.spec.ts:164]: The assertions for public/private keys and JWKs are repetitive across the tests. You could use a helper function to validate the structure of a key pair to improve readability, WDYT @forkimenjeckayang. ?

**Review by Ogenbertrand — CHANGES_REQUESTED** [review]

**Comment by Ogenbertrand** [comment]
Add tests for scenarios where invalid input is passed to generateKeyPairs, such as negative numbers or zero.

File: packages/multiple-did-identities/src/utils/generateKeyPairs.ts [hunk 2]
```diff
@@ -39,3 +39,95 @@ export const generateKeyPairs = async (numKeys: number) => {
...
+    privateKeyJwk: PrivateKeyJWK;
+  }> = [];
+
+  for (let i = 0; i < numKeys; i++) {
+    // Generate private and public keys
+    const privateKey = ed25519.utils.randomPrivateKey();
+    const publicKey = await ed25519.getPublicKey(privateKey);
+
+    // Create JWK for public and private keys
+    const publicKeyJwk: JWKKeys = {
```
Comment (Ogenbertrand) [inline packages/multiple-did-identities/src/utils/generateKeyPairs.ts:59]: Instead of hardcoding the structure of publicKeyJwk and privateKeyJwk, you could dynamically validate them against a schema right ?.

File: packages/multiple-did-identities/src/utils/generateKeyPairs.ts [hunk 3]
```diff
@@ -39,3 +39,95 @@ export const generateKeyPairs = async (numKeys: number) => {
...
+    privateKeyJwk: PrivateKeyJWK;
+  }> = [];
+
+  for (let i = 0; i < numKeys; i++) {
+    // Generate an X25519 private key
+    const privateKey = x25519.utils.randomPrivateKey();
+    const publicKey = x25519.scalarMultBase(privateKey);
+
+    // Create JWK for public and private keys
+    const publicKeyJwk: JWKKeys = {
```
Comment (Ogenbertrand) [inline packages/multiple-did-identities/src/utils/generateKeyPairs.ts:105]: same thing here too.

**Review by Ogenbertrand — CHANGES_REQUESTED** [review]

File: packages/multiple-did-identities/src/__tests__/did-peer-2-service.spec.ts
Comment (Ogenbertrand) [inline packages/multiple-did-identities/src/__tests__/did-peer-2-service.spec.ts:1]: Isn't it alright to create test fixtures, to reduce reusability among the code base?

File: packages/multiple-did-identities/src/__tests__/did-peer-3-service.spec.ts
Comment (Ogenbertrand) [inline packages/multiple-did-identities/src/__tests__/did-peer-3-service.spec.ts:1]: I think implementing test fixtures in the test base will help reduce usability. WDYT its up to you to do this when you find time right!.

**Comment by hugoib** [comment]
It looks good! I added a comment that will be necessary to address so the did peers will be able to be used in further steps. (For packing the message, the id of the private key should have the previously shown format). Thank you!

**Comment by Ogenbertrand** [comment]
LGTM.

**Comment by Ogenbertrand** [comment]
It can be close after the response from @hugoib.

File: packages/multiple-did-identities/src/tests/testFixtures.ts [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+          uri: 'http://example.com/didcomm',
+          accept: ['didcomm/v2'],
+          routingKeys: [],
+        },
+      },
+    ],
+  },
+  privateKeyV: {
+    id: `#key-1`,
+    type: 'JsonWebKey2020',
```
Comment (IngridPuppet) [inline packages/multiple-did-identities/src/tests/testFixtures.ts:87]: Looks like there are no tests for the structure of the id.

**Review by IngridPuppet — APPROVED** [review]
